lib/rubocop/cop/root_cops/factory_file_name.rb

```diff
@@ -0,0 +1,34 @@
+require "pry"
+
+module RuboCop
+  module Cop
+    module RootCops
+      class FactoryFileName < Cop
+        FILE_NAME_ERROR = "Factory name should match file name".freeze
+
+        def investigate(processed_source)
+          file_path = processed_source.buffer.name
+          @base_file_name = File.basename(file_path, ".rb")
+        end
+
+        def on_send(node)
+          method_name = node.method_name
+          return unless _factory_method?(method_name)
+
+          factory_symbol_name = *node.arguments[0]
+          factory_name = factory_symbol_name[0].id2name.split("__").first
```

Comment:
Should this be one underscore instead of two?

Reply:
This is in reference to [Dan's comment](https://github.com/Root-App/root-ruby-style/pull/9#discussion_r272742021)